contracts/Removal.sol

```diff
@@ -74,6 +105,12 @@ contract Removal is
     }
     _latestTokenId = ids[ids.length - 1] + 1;
     super.mintBatch(to, ids, amounts, data);
+    emit BatchMinted(ids);
+
+    setApprovalForAll(to, _msgSender(), true); // todo look at vesting contract for potentially better approach
+    if (decodedData.list) {
+      safeBatchTransferFrom(to, decodedData.marketAddress, ids, amounts, data);
```

Comment:
I wonder if we consider just baking the market address in during initialization?    I like the flexibility of being able to specify it but we don't have a strong usecase for that yet.  And we'd need to add the whitelisting mechanism for those market addresses.

Reply:
@jaycenhorton whatchya think?